src/replication_core.rs

```diff
@@ -20,57 +20,88 @@ impl Plugin for ReplicationCorePlugin {
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum SendPolicy {
+    /// Unreliable and Unordered
+    Unreliable,
+    /// Reliable and Unordered
+    Unordered,
+    /// Reliable and Ordered
+    Ordered,
+}
+
 /// A resource to create channels for [`bevy_renet::renet::ConnectionConfig`]
 /// based on number of added server and client events.
-#[derive(Clone, Copy, Default, Resource)]
+#[derive(Clone, Resource)]
 pub struct NetworkChannels {
-    /// Increments with each server event registration.
-    server: u8,
-    /// Increments with each client event registration.
-    client: u8,
+    /// Grows with each server event registration.
+    server: Vec<SendPolicy>,
+    /// Grows with each client event registration.
+    client: Vec<SendPolicy>,
+}
+
+impl Default for NetworkChannels {
+    fn default() -> NetworkChannels {
+        NetworkChannels {
+            server: Vec::with_capacity(u8::MAX as usize),
+            client: Vec::with_capacity(u8::MAX as usize),
```

Comment:
I would avoid preallocation with `u8::MAX`. Most of the time you won't use all events.

Reply:
My suggestion was to avoid preallocation at all because we don't know the exact number of events being used. And using random number without profiling doesn't make much sense. Let's just use the default derive for simplicity, that's what Bevy do.